server/src/main/java/org/jboss/as/server/deployment/module/descriptor/SharedMountHandle.java

```diff
@@ -0,0 +1,55 @@
+package org.jboss.as.server.deployment.module.descriptor;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.jboss.as.server.deployment.module.TempFileProviderService;
+import org.jboss.vfs.VFS;
+import org.jboss.vfs.VirtualFile;
+
+class SharedMountHandle implements Closeable {
+    private static Map<VirtualFile, SharedMountHandle> mountPoints = new HashMap<VirtualFile, SharedMountHandle>();
```

Comment:
This static map makes me uncomfortable. In most cases such things are wrong, and something that has a more limited scope is better; i.e. something controlled by a service that can be cleaned up when that service stops, or simple gc'd when the service is no longer referenced.

I'm not familiar enough with all this to suggest a service that can do this though.